test/extensions/filters/http/lua/lua_integration_test.cc

```diff
@@ -1331,5 +1331,38 @@ name: lua
   EXPECT_TRUE(response->complete());
 }
 
+// Test upstream host override functionality in Lua filter
+TEST_P(LuaIntegrationTest, UpstreamHostOverride) {
+  const std::string FILTER_AND_CODE = R"EOF(
+name: lua
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
+  default_source_code:
+    inline_string: |
+      function envoy_on_request(request_handle)
+        request_handle:setUpstreamOverrideHost("192.168.21.13", false)
+      end
+)EOF";
+
+  initializeFilter(FILTER_AND_CODE);
+
+  codec_client_ = makeHttpConnection(makeClientConnection(lookupPort("http")));
+  Http::TestRequestHeaderMapImpl request_headers{{":method", "GET"},
+                                                 {":path", "/test/long/url"},
+                                                 {":scheme", "http"},
+                                                 {":authority", "host"},
+                                                 {"x-forwarded-for", "10.0.0.1"}};
+
+  auto response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  waitForNextUpstreamRequest();
+
+  upstream_request_->encodeHeaders(default_response_headers_, true);
+  ASSERT_TRUE(response->waitForEndStream());
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().getStatusValue());
+
+  cleanup();
+}
```

Comment:
I think this test may make no big sense because the host cannot works anyway?

Reply:
Yeah, agreed that it's not very useful. I'll just remove it